cni-plugin/pkg/dataplane/linux/dataplane_linux.go

```diff
@@ -343,9 +345,18 @@ func SetupRoutes(hostNlHandle *netlink.Handle, hostVeth netlink.Link, result *cn
 			// Route already exists, but not necessarily pointing to the same interface.
 			case syscall.EEXIST:
 				// List all the routes for the interface.
-				routes, err := hostNlHandle.RouteList(hostVeth, netlink.FAMILY_ALL)
-				if err != nil {
-					return fmt.Errorf("error listing routes")
+				retries := 3
```

Comment:
Couldn't this be `RouteListRetryEINTR`? It looks very similar at least.

Reply:
```RouteListRetryEINTR``` directly calls ```netlink.RouteList```, whereas this one is via the handle.

Reply:
Could we provide an optional argument to provide the handle? Or maybe just require that all callers pass in a handle to use? Feels worthwhile to be consistent

Reply:
In other words, if we're going to define a library to help us with retries we should make it capable of handling all of the retry scenarios we need.